org.eclipse.cdt.cross.arc.gnu/src/com/arc/cdt/toolchain/tcf/TcfContent.java

```diff
@@ -183,6 +184,31 @@ public static TcfContent readFile(File f, String cpu) throws TcfContentException
         return tcfContent;
     }
 
+    /*
+     * It is needed because in java.util.Properties key termination characters are '=', ':' and
+     * whitespace. So if our TCF has several option like "--param ...", --param will be considered a
+     * key and therefore Properties will load only one of these options. If we escape a whitespace,
+     * it will be considered part of a key.
+     */
+    private static String escapeWhitespaces(String source) {
+        StringBuilder builder = new StringBuilder();
+        String[] lines = source.split("\n");
+        for (String line: lines) {
+            line = line.trim();
+            if (!line.isEmpty()) {
+                String[] parts = line.split(" ");
+                for (int i = 0; i < parts.length; i++) {
+                    if (i > 0) {
+                        builder.append("\\ ");
+                    }
```

Comment:
Besides I already told Anna to rewrite this with string.replaceAll() or something like that :)